core/src/main/scala/io/gearpump/serializer/FastKryoSerializer.scala

```diff
@@ -74,7 +76,18 @@ class FastKryoSerializer(system: ExtendedActorSystem) {
   }
 
   def deserialize(msg : Array[Byte]): AnyRef = {
-      kryoSerializer.fromBinary(msg)
+    kryoSerializer.fromBinary(msg)
+  }
+
+  private def getDelegator(): Option[SerializationDelegate] = {
+    Try(config.getString(Constants.GEARPUMP_SERIALIZER_DELEGETE_FACTORY)).map { factoryClass =>
+      if (factoryClass != "") {
+        val factory = Class.forName(factoryClass).newInstance()
+        Some(factory.asInstanceOf[IDelegateFactory].getDelegate(system))
```

Comment:
Suppose user provide a non-IDelegateFactory class, the getDelegtor function will throw exception. If we fall back to non-delegator, user's message still can not be serialized correctly, so maybe throw the exception earlier  is better.